src/actors/cficaches.rs

```diff
@@ -0,0 +1,209 @@
+    Parsing,
+
+    #[fail(display = "failed to parse object")]
+    ObjectParsing,
+
+    #[fail(display = "symcache building took too long")]
+    Timeout,
+}
+
+symbolic::common::derive_failure!(
+    CfiCacheError,
+    CfiCacheErrorKind,
+    doc = "Errors happening while generating a symcache"
+);
+
+impl From<io::Error> for CfiCacheError {
+    fn from(e: io::Error) -> Self {
+        e.context(CfiCacheErrorKind::Io).into()
+    }
+}
+
+pub struct CfiCacheActor {
+    symcaches: Addr<CacheActor<FetchCfiCacheInternal>>,
+    objects: Addr<ObjectsActor>,
+    threadpool: Arc<ThreadPool>,
+}
+
+impl Actor for CfiCacheActor {
+    type Context = Context<Self>;
+}
+
+impl CfiCacheActor {
+    pub fn new(
+        symcaches: Addr<CacheActor<FetchCfiCacheInternal>>,
+        objects: Addr<ObjectsActor>,
+        threadpool: Arc<ThreadPool>,
+    ) -> Self {
+        CfiCacheActor {
+            symcaches,
+            objects,
+            threadpool,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct CfiCacheFile {
+    inner: Option<ByteView<'static>>,
+    scope: Scope,
+    request: FetchCfiCacheInternal,
+}
+
+impl CfiCacheFile {
+    pub fn parse(&self) -> Result<Option<minidump::cfi::CfiCache<'_>>, CfiCacheError> {
+        let bytes = match self.inner {
+            Some(ref x) => x,
+            None => return Ok(None),
+        };
+
+        if &bytes[..] == b"malformed" {
+            return Err(CfiCacheErrorKind::ObjectParsing.into());
+        }
+
+        Ok(Some(
+            minidump::cfi::CfiCache::from_bytes(bytes.clone())
+                .context(CfiCacheErrorKind::Parsing)?,
+        ))
+    }
+}
+
+#[derive(Clone)]
+pub struct FetchCfiCacheInternal {
+    request: FetchCfiCache,
+    objects: Addr<ObjectsActor>,
+    threadpool: Arc<ThreadPool>,
+}
+
+impl CacheItemRequest for FetchCfiCacheInternal {
+    type Item = CfiCacheFile;
+    type Error = CfiCacheError;
+
+    fn get_cache_key(&self) -> CacheKey {
+        let mut cache_key = self.request.identifier.cache_key();
+
+        for source in &self.request.sources {
+            cache_key.push_str(&format!(".s:{}", source.id()));
+        }
+
+        CacheKey {
+            cache_key,
+            scope: self.request.scope.clone(),
+        }
+    }
+
+    fn compute(&self, path: &Path) -> Box<dyn Future<Item = Scope, Error = Self::Error>> {
+        let objects = self.objects.clone();
+
+        let path = path.to_owned();
+        let threadpool = self.threadpool.clone();
+
+        // TODO: Backoff + retry when download is interrupted? Or should we just have retry logic
+        // in Sentry itself?
+        let result = objects
+            .send(FetchObject {
+                filetypes: FileType::from_object_type(&self.request.object_type),
+                identifier: self.request.identifier.clone(),
+                sources: self.request.sources.clone(),
+                scope: self.request.scope.clone(),
+                // TODO: Prefer files with unwind info
```

Comment:
@jan-auer right now fetching objects for cfi caches goes through the same heuristic that is used for fetching files for symcaches, i.e. it prefers objects where `has_debug_info() == true`. I assume this needs to be changed to prefer objects where `has_unwind_info() == true` instead?

Reply:
Correct. See the „Symbol Lookup Strategy“ document for a detailed table.